packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -29,71 +29,124 @@ import {
+	 * navigation work, avoiding undesired behaviors.
+	 * For instance, it will block to move between menu items
+	 * when the LinkOver is focused.
+	 *
+	 * @param {Event} event
+	 */
+	const handleLinkControlOnKeyDown = ( event ) => {
+		const { keyCode } = event;
 
-	const onKeyDown = ( event ) => {
-		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
+		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( keyCode ) > -1 ) {
 			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
 			event.stopPropagation();
 		}
 	};
 
-	const closeURLPopover = () => {
-		setIsEditingLink( false );
-		setUrlInput( null );
-		setIsLinkOpen( false );
-	};
-
-	const autocompleteRef = useRef( null );
-
-	const onFocusOutside = ( event ) => {
-		const autocompleteElement = autocompleteRef.current;
-		if ( autocompleteElement && autocompleteElement.contains( event.target ) ) {
-			return;
+	/**
+	 * Updates the link attribute when it changes
+	 * through of the `onLinkChange` LinkControl callback.
+	 *
+	 * @param {Object|null} itemLink Link object if it has been selected. Otherwise, Null.
+	 */
+	const updateLink = ( itemLink ) => {
+		if ( ! itemLink ) {
+			// If not link, then set empty string to link attrs.
+			return setAttributes( { title: '', url: '' } );
 		}
-		closeURLPopover();
+
+		setAttributes( { title: itemLink.title, url: itemLink.url } );
 	};
 
-	const stopPropagation = ( event ) => {
-		event.stopPropagation();
+	/**
+	 * It updates the link attribute when the
+	 * link settings changes.
+	 *
+	 * @param {string} setting Setting type, for instance, `new-tab`.
+	 * @param {string} value Setting type value.
+	 */
+	const updateLinkSetting = ( setting, value ) => {
+		if ( 'new-tab' ) {
+			setAttributes( { opensInNewTab: value } );
+		}
 	};
 
-	const { label, url } = attributes;
 	let content;
 	if ( isSelected ) {
 		content = (
-			<TextControl
-				ref={ plainTextRef }
-				className="wp-block-navigation-menu-item__field"
-				value={ label }
-				onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
-				label={ __( 'Navigation Label' ) }
-				hideLabelFromVision={ true }
-			/>
+			<div className="wp-block-navigation-menu-item__field-container">
+				<TextControl
+					ref={ plainTextRef }
+					className="wp-block-navigation-menu-item__field"
+					value={ label }
+					onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
+					label={ __( 'Navigation Label' ) }
+					hideLabelFromVision={ true }
+				/>
+				{ isLinkOpen &&
+					<LinkControl
+						className="wp-block-navigation-menu-item__inline-link-input"
+						onKeyDown={ handleLinkControlOnKeyDown }
+						onKeyPress={ ( event ) => event.stopPropagation() }
+						currentLink={ link }
+						onLinkChange={ updateLink }
+						onClose={ () => {
+							setWasCloseByLinkControl( true );
+							setIsLinkOpen( false );
+						} }
+						currentSettings={ { 'new-tab': opensInNewTab } }
```

Comment:
Is it weird that we are using hyphens here for object keys? I think this is probably an issue in `LinkControl` right?

Reply:
Created issue https://github.com/WordPress/gutenberg/issues/18206